dandischema/models.py

```diff
@@ -133,6 +133,19 @@ def json_dict(self):
         """
         return json.loads(self.json(exclude_none=True, cls=HandleKeyEnumEncoder))
 
+    @validator("schemaKey", always=True)
+    def ensure_schemakey(cls, val):
+        tempval = val
+        if "Published" in cls.__name__:
+            tempval = "Published" + tempval
+        if "BareAsset" == cls.__name__:
+            tempval = "Bare" + tempval
```

Comment:
so it could make it `BarePublished<val>`?  shouldn't `Published` be added later so it would be `PublishedBare<val>`?

Reply:
that should not happen here. the second if is an equality check, in which case the only value tempval should have is `Asset`, thus turning it into `BareAsset`. anything else it should throw an error in the next few lines.